state/execution.go

```diff
@@ -91,23 +91,83 @@ func (blockExec *BlockExecutor) SetEventBus(eventBus types.BlockEventPublisher)
 
 // CreateProposalBlock calls state.MakeBlock with evidence from the evpool
 // and txs from the mempool. The max bytes must be big enough to fit the commit.
-// Up to 1/10th of the block space is allcoated for maximum sized evidence.
+// Up to 1/10th of the block space is allocated for maximum sized evidence.
 // The rest is given to txs, up to the max gas.
 func (blockExec *BlockExecutor) CreateProposalBlock(
 	height int64,
 	state State, commit *types.Commit,
 	proposerAddr []byte,
+	createBlockFromApp bool,
 ) (*types.Block, *types.PartSet) {
+	var txs types.Txs
+	var timestamp time.Time
+	evidence := blockExec.evpool.PendingEvidence(state.ConsensusParams.Evidence.MaxNum)
+	if createBlockFromApp {
+		voteInfos := make([]abci.VoteInfo, commit.Size())
```

Comment:
we may want to abstract this part of code away as it's essentially the same logic in`BeginBlock`

Reply:
do it in the second PR